test/functional/p2p_getaddr_caching.py

```diff
@@ -63,17 +41,16 @@ def set_test_params(self):
 
     def run_test(self):
         self.log.info('Create connection that sends and requests addr messages')
-        addr_source = self.nodes[0].add_p2p_connection(P2PInterface())
+        self.nodes[0].add_p2p_connection(P2PInterface())
 
-        msg_send_addrs = msg_addr()
         self.log.info('Fill peer AddrMan with a lot of records')
         # Since these addrs are sent from the same source, not all of them will be stored,
         # because we allocate a limited number of AddrMan buckets per addr source.
-        total_addrs = 10000
-        addrs = gen_addrs(total_addrs)
-        for i in range(int(total_addrs/MAX_ADDR_TO_SEND)):
-            msg_send_addrs.addrs = addrs[i * MAX_ADDR_TO_SEND:(i + 1) * MAX_ADDR_TO_SEND]
-            addr_source.send_and_ping(msg_send_addrs)
+        for i in range(10000):
+            first_octet = i >> 8
+            second_octet = i % 256
+            a = "{}.{}.1.1".format(first_octet, second_octet)
+            self.nodes[0].addpeeraddress(a, 8333)
```

Comment:
Perhaps assert that `len(self.nodes[0].getnodeaddresses(0))` is > 4000 here (to ensure that a getaddr response has at least 1000 records in it)